src/Microsoft.Framework.DependencyInjection/ServiceProvider.cs

```diff
@@ -145,15 +145,12 @@ internal IServiceCallSite GetResolveCallSite(IService service, ISet<Type> callSi
 
         public void Dispose()
         {
-            var disposables = Interlocked.Exchange(ref _disposables, null);
```

Comment:
This looks like it was done to avoid a double-dispose if `Dispose()` is called concurrently. With the current code it won't double-dispose any services but could call `Clear()` twice - is this a concern?

I suspect we're not too concerned about this scenario in practice. If there's any code you have operating on an SP while it's being disposed, you already have a bug.